test/hipcub/test_hipcub_block_exchange.cpp

```diff
@@ -351,34 +353,62 @@ TYPED_TEST(HipcubBlockExchangeTests, BlockedToWarpStriped)
     std::vector<output_type> expected(size);
     std::vector<output_type> output(size, output_type(0));
 
-    constexpr size_t warp_size = test_utils::get_min_warp_size(block_size, size_t(HIPCUB_WARP_THREADS));
-    constexpr size_t warps_no = (block_size + warp_size - 1) / warp_size;
-    constexpr size_t items_per_warp = warp_size * items_per_thread;
+    constexpr size_t warp_size_32 = test_utils::get_min_warp_size(block_size, size_t(HIPCUB_WARP_SIZE_32));
+    constexpr size_t warp_size_64 = test_utils::get_min_warp_size(block_size, size_t(HIPCUB_WARP_SIZE_64));
+    constexpr size_t warps_no_32 = (block_size + warp_size_32 - 1) / warp_size_32;
+    constexpr size_t warps_no_64 = (block_size + warp_size_64 - 1) / warp_size_64;
+    constexpr size_t items_per_warp_32 = warp_size_32 * items_per_thread;
+    constexpr size_t items_per_warp_64 = warp_size_64 * items_per_thread;
 
     // Calculate input and expected results on host
     std::vector<type> values(size);
     std::iota(values.begin(), values.end(), 0);
-    for(size_t bi = 0; bi < size / items_per_block; bi++)
+    if (current_device_warp_size == HIPCUB_WARP_SIZE_32)
```

Comment:
Probably we could reduce the code duplication here, and we can get ridd off this if else statement here, if we use the runtime variables.